sdk/core/core/inc/az_pair.h

```diff
@@ -4,39 +4,34 @@
 #ifndef AZ_PAIR_H
 #define AZ_PAIR_H
 
-#include <az_callback.h>
+#include <az_action.h>
 #include <az_contract.h>
 #include <az_span.h>
 
 #include <_az_cfg_prefix.h>
 
 /// A pair of strings.
 typedef struct {
-  az_const_span key;
-  az_const_span value;
+  az_span key;
+  az_span value;
 } az_pair;
 
 /// A span of pairs.
-typedef struct {
-  az_pair const * begin;
-  size_t size;
-} az_pair_span;
+AZ_SPAN_TYPE(az_pair_span, az_pair const)
```

Comment:
do these new types have the same "methods" as the regular span and mut span? if not I strongly suggest, again writing out the type here. If so you're using macros to implement c++ templates and that doesn't tend to end that well.